src/precice/SolverInterface.hpp

```diff
@@ -147,10 +165,28 @@ class SolverInterface {
   /**
    * @brief Finalizes preCICE.
    *
+   * If initialize() has been called:
+   *
+   * - Synchronizes with remote participants
+   * - handles final exports
+   * - cleans up general state
+   *
+   * Always:
+   *
+   * - flushes and finalizes Events
+   * - finalizes managed PETSc
+   * - finalizes managed MPI
+   *
    * @pre finalize() has not been called.
    *
    * @post Communication channels are closed.
    * @post Meshes and data are deallocated
+   * @post Finalized managed PETSc
+   * @post Finalized managed MPI
+   *
+   * @warning
+   * Finalize is not the inverse of initialize().
+   * Finalize has to be called after construction.
```

Comment:
It will be called in the destructor if it wasn't called.

Reply:
Then I will remove these lines. I don't think we really need the warnings here.

Reply:
6a42ccbe756f69cfdda319a034bf02951da11a55